plugins/linear-genome-view/src/BaseLinearDisplay/models/TrackHeightMixin.tsx

```diff
@@ -26,11 +29,49 @@ export default function TrackHeightMixin() {
        * #property
        */
       scrollTop: 0,
+      /**
+       * #property
+       */
+      firstRenderHeight: 100,
+    }))
+    .actions(self => ({
+      /**
+       * #action
+       */
+      setFirstRenderHeight(val: number) {
+        self.firstRenderHeight = val
+        return val
+      },
     }))
     .views(self => ({
-      get height() {
+      /**
+       * #getter
+       * returns the height value as determined by the trackHeightSetting
+       */
+      get adjustLayout() {
+        const { trackHeightSetting } = getContainingView(
+          self,
+        ) as LinearGenomeViewModel
+
         // @ts-expect-error
-        return self.heightPreConfig ?? (getConf(self, 'height') as number)
+        const height = self.layoutMaxHeight
+
+        return trackHeightSetting === 'on'
+          ? height
+          : trackHeightSetting === 'first_render' &&
+              self.firstRenderHeight === 100
+            ? self.setFirstRenderHeight(height)
+            : trackHeightSetting === 'first_render'
+              ? self.firstRenderHeight
```

Comment:
Having trouble with some logic here, but not sure it’s a game changer:
Technically, this will incorrectly re-size if the user’s track is max block height 100, and then they scroll to a block that is greater than 100.
The trouble is the height and consequently adjustLayout are called ~3 times while the blocks are being calculated, and before the actual height is determined, the height is the default of 100 (set as such so the track while loading doesn’t get weirdly small).
So, I don't have a "confident" way to determine that the track has been "rendered" properly before setting the firstRenderHeight.

@cmdcolin thoughts? Kind of a niche scenario (i.e. the track would have to be exactly a max height of 100 in the viewable area, and then have a greater height elsewhere)

Reply:
1. calling an action in a getter should likely be avoided. this code may want to be combined with an autorun or something like this
2. I also don't have a confident way to determine that a track has been rendered for the first time. likely creating that some code/infrastructure to detect this would be valuable though. the 3 times is just for all the static blocks
3. not sure i fully understand the scenario you describe, but i would try to avoid using just a 'magic variable' in the code like 100 and overgitting to it